apps/site/assets/ts/schedule/components/schedule-finder/ScheduleFinderForm.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { FormEvent, ReactElement, useState } from "react";
+import { DirectionId, DirectionInfo, Route } from "../../../__v3api";
+import { SimpleStopMap, SelectedOrigin } from "../__schedule";
+import icon from "../../../../static/images/icon-schedule-finder.svg";
+import renderSvg from "../../../helpers/render-svg";
+import SelectContainer from "./SelectContainer";
+
+const validDirections = (directionInfo: DirectionInfo): DirectionId[] =>
+  ([0, 1] as DirectionId[]).filter(dir => directionInfo[dir] !== null);
+
+interface Props {
+  onDirectionChange: (direction: DirectionId) => void;
+  onOriginChange: (origin: SelectedOrigin) => void;
+  onOriginSelectClick: () => void;
+  onSubmit?: () => void;
+  route: Route;
+  selectedDirection: DirectionId;
+  selectedOrigin: SelectedOrigin;
+  stopsByDirection: SimpleStopMap;
+}
+
+export default ({
+  onDirectionChange,
+  onOriginChange,
+  onSubmit = () => {},
+  onOriginSelectClick,
+  route,
+  selectedDirection,
+  selectedOrigin,
+  stopsByDirection
+}: Props): ReactElement => {
+  const {
+    direction_names: directionNames,
+    direction_destinations: directionDestinations
+  } = route;
+
+  const [originError, setOriginError] = useState(false);
+
+  const handleOriginClick = (): void => {
+    setOriginError(false);
+    onOriginSelectClick();
+  };
+
+  const handleSubmit = (event: FormEvent): void => {
+    event.preventDefault();
+
+    if (selectedOrigin === null) {
+      setOriginError(true);
+    } else {
+      setOriginError(false);
+      onSubmit();
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <h2 className="h3 schedule-finder__heading">
+        {renderSvg("c-svg__icon", icon)} Schedule Finder
+      </h2>
+
+      <div className="schedule-finder__prompt">
+        Choose a stop to get schedule information and real-time departure
+        predictions.
+      </div>
+
+      {originError && (
+        <div className="error-container">
+          <span role="alert">Please provide an origin</span>
+        </div>
+      )}
+
+      <div className="schedule-finder__inputs">
+        <label className="schedule-finder__label">
+          Choose a direction
+          <SelectContainer>
```

Comment:
@digitalcora so this is a new usage to me, I assume that the elements inside the `SelectContainer` tag get bound to the `children` of the `SelectContainer` component itself? is that automatic or am I overlooking an explicit binding somewhere?

Reply:
The former. Anything you put inside the "tag" of a React component is passed to it as a `children` prop.